runtime/src/main/java/org/corfudb/runtime/clients/LogUnitClient.java

```diff
@@ -172,14 +174,16 @@ private static Object handleReadDataCorruption(CorfuPayloadMsg<ReadResponse> msg
      */
     public CompletableFuture<Boolean> write(long address, Set<UUID> streams, long rank,
                                             Object writeObject, Map<UUID, Long> backpointerMap) {
+        Timer.Context context = getTimerContext("writeObject");
         ByteBuf payload = ByteBufAllocator.DEFAULT.buffer();
         Serializers.CORFU.serialize(writeObject, payload);
         WriteRequest wr = new WriteRequest(WriteMode.NORMAL, null, payload);
         wr.setStreams(streams);
         wr.setRank(rank);
         wr.setBackpointerMap(backpointerMap);
         wr.setGlobalAddress(address);
-        return router.sendMessageAndGetCompletable(CorfuMsgType.WRITE.payloadMsg(wr));
+        CompletableFuture<Boolean> cf = router.sendMessageAndGetCompletable(CorfuMsgType.WRITE.payloadMsg(wr));
+        return context == null ? cf : cf.thenApply(x -> { context.stop(); return x; });
```

Comment:
this turns the write call synchronous. probably not a desirable side-effect of metering we want to tolerate.